src/main/java/reposense/util/FileUtil.java

```diff
@@ -82,15 +82,12 @@ public static void zip(Path sourcePath, Path outputPath, String... fileTypes) {
         ) {
             Set<Path> allFiles = getFilePaths(sourcePath, fileTypes);
             for (Path path : allFiles) {
+                String filePath = sourcePath.relativize(path.toAbsolutePath()).toString();
+                String zipEntry = Files.isDirectory(path) ? filePath + File.separator : filePath;
                 try (InputStream is = Files.newInputStream(path)) {
```

Comment:
This will cause `IOException` since we are trying to open an `InputStream` for a directory, thus resulting in an empty zip being generated. You should do the directory check outside and skip this portion.

Reply:
`IOException` will be caused if you try and write something to `ZipOutStream` after putting the entry. I'm checking if its a regular file before to the `ZipOutputStream`.

Reply:
Your implementation will fail on Windows. 

Try:
```
Set<Path> allFilePaths = getFilePaths(sourcePath, fileTypes);
    for (Path filePath : allFilePaths) {
        String path = sourcePath.relativize(filePath.toAbsolutePath()).toString();
        String zipEntryPath = Files.isDirectory(filePath) ? path + "/" : path;
        zos.putNextEntry(new ZipEntry(zipEntryPath.replace("\\", "/")));

        if (Files.isRegularFile(filePath)) {
            try (InputStream is = Files.newInputStream(filePath)) {
                int length;
                while ((length = is.read(buffer.array())) > 0) {
                    zos.write(buffer.array(), 0, length);
                }
           }
       }

     zos.closeEntry();
}
```